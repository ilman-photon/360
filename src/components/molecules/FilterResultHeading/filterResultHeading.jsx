import { Box } from "@mui/system";
import constants from "../../../utils/constants";
import { StyledButton } from "../../atoms/Button/button";
import TuneIcon from "@mui/icons-material/Tune";
import KeyboardArrowDownIcon from "@mui/icons-material/KeyboardArrowDown";
import ArrowBackIosIcon from "@mui/icons-material/ArrowBackIos";
import ArrowForwardIosIcon from "@mui/icons-material/ArrowForwardIos";
import styles from "./filterResultHeading.module.scss";
import CloseIcon from "@mui/icons-material/Close";
import FilterBy from "../FilterBy/filterBy";
import { useEffect, useState } from "react";
import { Button, IconButton, Stack, Typography } from "@mui/material";
import LocationOnOutlinedIcon from "@mui/icons-material/LocationOnOutlined";
import { colors } from "../../../styles/theme";
import Image from "next/image";
import FilterHeadingFilled from "../FilterHeading/filterHeadingFilled";
import { getDates, isPrevArrowDisable } from "../../../utils/appointment";

export const FilterResultHeading = ({
  appliedFilter = [],
  numberFilter = 30,
  isDesktop = false,
  isTablet = false,
  filterData = {
    location: "New York, NY",
    date: "",
    purposeOfVisit: "Eye exam",
    insuranceCarrier: "Aethna",
  },
  onSearchProvider = () => {
    // This is intentional
  },
  onNextScheduleClicked = () => {
    // This is intentional
  },
  onPrevScheduleClicked = () => {
    // This is intentional
  },
  purposeOfVisitData = [],
  insuranceCarrierData = [],
  rangeDate = { startDate: "", endDate: "" },
  filter = [],
  onActivFilter,
  title = "",
  subtitle = "",
  isGeolocationEnabled = false,
  onChangeLocation = () => {
    // This is intentional
  },
  currentCity = "",
}) => {
  const imageSrcState = "/searchInputIcon.png";
  const imageSrcFilled = "/searchFilledIcon.png";
  const { APPOINTMENT_TEST_ID } = constants.TEST_ID;
  const weekColumn = [
    "mondaySchedule",
    "tuesdaySchedule",
    "wednesdaySchedule",
    "thusdaySchedule",
    "fridaySchedule",
    "saturdaySchedule",
  ];
  const [filterOpen, setFilterOpen] = useState(false);
  const [dialogOpen, setDialogOpen] = useState(false);
  const [activeFilter, setActiveFilter] = useState([]);
  const [dateList, setDateList] = useState({
    dateRange: [],
    dateListName: [],
  });

  useEffect(() => {
    const newDateList = getDates(
      new Date(rangeDate.startDate),
      new Date(rangeDate.endDate)
    );
    if (rangeDate.startDate && rangeDate.endDate) {
      setDateList(newDateList);
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [rangeDate]);

  function renderAppliedFilter() {
    return appliedFilter.map((option, idx) => {
      return (
        <Box className={styles.filterChildButton} key={idx} tabIndex="0">
          {option.name}
          <IconButton
            // eslint-disable-next-line jsx-a11y/aria-props
            aria-description="Close"
            onClick={() => {
              const id = appliedFilter.findIndex((x) => x.name === option.name);
              if (id > -1) {
                const data = [...appliedFilter];
                data.splice(id, 1);
                setActiveFilter([...data]);
                onActivFilter([...data]);
              }
            }}
          >
            <CloseIcon className={styles.closeIcon} />
          </IconButton>
        </Box>
      );
    });
  }

  const onSetFilter = (newFilterData) => {
    setFilterOpen(!filterOpen);
    setActiveFilter(newFilterData);
    onActivFilter(newFilterData);
  };

  function handleCloseDialog() {
    //Reset data when cancel the dialog
    setDialogOpen(false);
  }

  function handleOpenDialog() {
    //Reset data when cancel the dialog
    setDialogOpen(true);
  }

  function renderDesktopView() {
    return (
      <Box
        className={[
          styles.filterContainer,
          isTablet ? styles.isTablet : null,
        ].join(" ")}
      >
        <Box className={styles.filterButtonContainer}>
          <FilterBy
            activedFilter={[...appliedFilter]}
            filter={filter}
            isOpen={filterOpen}
            onClose={() => {
              setFilterOpen(!filterOpen);
            }}
            onDone={(selectedFilterData) => {
              onSetFilter(selectedFilterData);
            }}
          ></FilterBy>
          <StyledButton
            className={styles.filterButton}
            mode={constants.SECONDARY}
            size={constants.SMALL}
            gradient={false}
            data-testid={APPOINTMENT_TEST_ID.filterbtn}
            onClick={() => {
              setFilterOpen(!filterOpen);
            }}
          >
            <TuneIcon className={styles.tuneIcon} />
            Filters
            <KeyboardArrowDownIcon className={styles.keydownIcon} />
          </StyledButton>
          <Box display="flex" flexWrap="wrap">
            {renderAppliedFilter()}
          </Box>
        </Box>
        <Box
          sx={{
            marginTop: "16px",
            display: "grid",
            justifyContent: "space-between",
            alignContent: "space-between",
            gridTemplateRows: "auto",
            gridTemplateAreas: `"filterDetails calenderDetails"`,
          }}
          className={styles.filterHeaderCalendar}
        >
          <Box
            sx={{ gridArea: "filterDetails" }}
            className={styles.filterFoundWarpper}
          >
            <Typography className={styles.filterFound} tabindex={"0"}>
              {`${numberFilter} In-network providers`}
            </Typography>
          </Box>
          <Box
            sx={{
              gridArea: "calenderDetails",
              display: "grid",
              gridTemplateColumns: "3% 94% 3%",
              gridTemplateRows: "auto",
              gridTemplateAreas: `"arrowLeft caledar arrowRight"`,
            }}
            className={styles.dateListContainer}
          >
            <Button
              role={"button"}
              onClick={() => {
                if (!isPrevArrowDisable(dateList)) {
                  const date = new Date(dateList.dateRange[0]);
                  date.setDate(date.getDate() - 7);
                  onPrevScheduleClicked("week", date);
                }
              }}
              sx={{
                gridArea: "arrowLeft",
                width: "22px",
                minWidth: "22px",
                padding: 0,
              }}
              className={
                isPrevArrowDisable(dateList)
                  ? styles.prevArrowDisable
                  : styles.prevArrowActive
              }
              aria-label={"Navigate to previous week option"}
            >
              <ArrowBackIosIcon
                sx={{
                  margin: "auto",
                  width: "22px",
                }}
              />
            </Button>

            <Box
              sx={{
                gridArea: "caledar",
                display: "grid",
                gridTemplateColumns: "repeat(6, 1fr)",
                justifyContent: "center",
                alignContent: "center",
                gridTemplateRows: "auto",
                gridTemplateAreas: `"mondaySchedule tuesdaySchedule wednesdaySchedule thusdaySchedule fridaySchedule saturdaySchedule"`,
              }}
              className={styles.calendarHeading}
            >
              {weekColumn.map((option, idx) => {
                return (
                  <Box
                    key={idx}
                    sx={{ gridArea: option, textAlign: "center" }}
                    className={styles.calenderDayWarpper}
                  >
                    <Typography
                      className={styles.calenderDay}
                      aria-label={`${option.slice(0, -8)}`}
                      tabIndex={"0"}
                    >
                      {option.slice(0, 3)}
                    </Typography>
                    <Typography
                      className={styles.calenderMonth}
                      aria-label={`${dateList.dateListName[idx]}`}
                      tabIndex={"0"}
                    >
                      {dateList.dateListName[idx]}
                    </Typography>
                  </Box>
                );
              })}
            </Box>
            <Button
              role={"button"}
              onClick={() => {
                const date = new Date(dateList.dateRange[5]);
                date.setDate(date.getDate() + 7);
                onNextScheduleClicked("week", date);
              }}
              sx={{
                gridArea: "arrowRight",
                width: "22px",
                minWidth: "22px",
                padding: 0,
              }}
              className={styles.prevArrowActive}
              aria-label={"Navigate to next week option"}
            >
              <ArrowForwardIosIcon
                sx={{
                  margin: "auto",
                  width: "22px",
                  cursor: "pointer",
                }}
              />
            </Button>
          </Box>
        </Box>
      </Box>
    );
  }

  function renderMobileView() {
    let renderSubFilter = filterData.purposeOfVisit
      ? `${filterData.purposeOfVisit} â€¢ `
      : "";
    renderSubFilter += filterData.insuranceCarrier?.name || "";
    return (
      <Box
        className={styles.mobileFilterContainer}
        sx={{
          backgroundColor: title && subtitle ? "transparent" : colors.darkGreen,
          height: title && subtitle ? "165px" : "105px",
        }}
        aria-live="polite"
      >
        <FilterBy
          activedFilter={[...activeFilter]}
          filter={filter}
          isOpen={filterOpen}
          onClose={() => {
            setFilterOpen(!filterOpen);
          }}
          onDone={(selectedFilterData) => {
            onSetFilter(selectedFilterData);
          }}
        ></FilterBy>
        <Stack
          justifyContent={"center"}
          height={"100%"}
          paddingX={"14px"}
          className={title && subtitle ? styles.titleContainer : {}}
        >
          {title && subtitle && (
            <Stack className={styles.subtitleContainer}>
              <Typography className={styles.titleElement} tabindex={0}>
                {title}
              </Typography>
              <Typography className={styles.subtitleElement} tabindex={0}>
                {subtitle}
              </Typography>
            </Stack>
          )}
          <Stack
            className={
              title && subtitle
                ? styles.stackFilterContainer
                : styles.mobileFilterStyle
            }
            direction={"row"}
          >
            <LocationOnOutlinedIcon
              sx={{
                margin: "auto 14px",
                width: "25px",
                height: "25px",
                color: colors.darkGreen,
              }}
            />
            <Box
              className={styles.mobileFilterTitleContainer}
              data-testid={"open-filter-modal"}
              onClick={handleOpenDialog}
            >
              <Button
                aria-label={filterData.location}
                className={[
                  styles.mobileFilterStyles,
                  styles.mobileFilterButton,
                ].join(" ")}
              >
                <Typography
                  variant={"bodyMedium"}
                  className={styles.mobileFilterTitle}
                >
                  {filterData.location || "City, state, or zip code"}
                </Typography>
                {renderSubFilter}
              </Button>
            </Box>
            <Box
              className={styles.mobileFilterImageContainer}
              onClick={() => {
                setFilterOpen(true);
              }}
            >
              {!(title && subtitle) && (
                <Button
                  aria-label={"Filter"}
                  data-testId="filter-button-mobile"
                  sx={{
                    width: "31px",
                    minWidth: "31px",
                    padding: 0,
                  }}
                >
                  <Image
                    alt=""
                    src={
                      activeFilter.length > 0 ? imageSrcFilled : imageSrcState
                    }
                    width={31}
                    height={31}
                  />
                </Button>
              )}
            </Box>
          </Stack>
        </Stack>
        {
          <FilterHeadingFilled
            openDialog={dialogOpen}
            onCloseDialog={handleCloseDialog}
            filterData={filterData}
            onSearchProvider={onSearchProvider}
            purposeOfVisitData={purposeOfVisitData}
            insuranceCarrierData={insuranceCarrierData}
            isGeolocationEnabled={isGeolocationEnabled}
            onChangeLocation={onChangeLocation}
            currentCity={currentCity}
          />
        }
      </Box>
    );
  }

  return <>{isDesktop ? renderDesktopView() : renderMobileView()}</>;
};

export default FilterResultHeading;
