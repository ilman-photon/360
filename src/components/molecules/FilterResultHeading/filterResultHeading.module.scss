@import "../../../../styles/colors.scss";

.filterContainer {
  width: 1128px;

  &.isTablet {
    width: 100%;
  }
}

.filterButtonContainer {
  display: flex;

  .filterButton {
    width: 148px;
    height: 30px;
    border: 1px solid #205a63;
    border-radius: 24px;
    font-style: normal;
    font-weight: 600;
    font-size: 14px;
    line-height: 18px;
    justify-content: left;
  }
}

.filterChildButton {
  background: #ffffff;
  border: 1px solid #003b4a;
  border-radius: 100px;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 3px 4px 3px 10px;
  margin-left: 10px;
  font-style: normal;
  font-weight: 500;
  font-size: 14px;
  line-height: 17px;
  color: #6d6f73;
  .closeIcon {
    width: 15px;
    height: 15px;
    margin-left: 5px;
  }
}

.tuneIcon {
  color: $iconBlack;
}

.keydownIcon {
  position: absolute;
  left: 72%;
  color: rgba(0, 0, 0, 0.54);
}

.filterFoundWarpper {
  .filterFound {
    font-family: "Roboto", sans-serif;
    font-style: normal;
    font-weight: 600;
    font-size: 16px;
    line-height: 24px;
    color: $darkGreen;
  }
}

.calenderDayWarpper {
  .calenderDay {
    font-family: "Roboto", sans-serif;
    font-style: normal;
    font-weight: 400;
    font-size: 15px;
    line-height: 17px;
    text-align: center;
    color: $darkGreen;
    text-transform: capitalize;
  }
  .calenderMonth {
    font-family: "Roboto", sans-serif;
    font-style: normal;
    font-weight: 600;
    font-size: 15px;
    line-height: 17px;
    text-align: center;
    color: $darkGreen;
    text-transform: capitalize;
  }
}

.mobileFilterContainer {
  width: 100%;

  .titleContainer {
    position: relative;
    background: linear-gradient(96.59deg, #003b4a 10.15%, #0389ab 140.82%);
  }
  .subtitleContainer {
    margin-bottom: 30px;
  }
  .titleElement {
    font-family: "Inter", sans-serif;
    font-style: normal;
    font-weight: 500;
    font-size: 26px;
    line-height: 32px;
    color: #ffffff;
  }
  .subtitleElement {
    font-family: "Libre Franklin", sans-serif;
    font-style: normal;
    font-weight: 400;
    font-size: 18px;
    line-height: 26px;
    letter-spacing: 0.0016em;
    margin-top: 8px;
    color: #ffffff;
  }
  .stackFilterContainer {
    height: 56px;
    background: #fff;
    border-radius: 47px;
    border: 1px solid #bdbdbd;
    box-shadow: 0px 1px 5px rgb(0 0 0 / 18%);
    position: absolute;
    width: 94%;
    top: 84%;
  }
  .mobileFilterStyle {
    height: 56px;
    background: #fff;
    border: 1px solid $teal;
    border-radius: 47px;
  }
}

.mobileFilterTitle {
  color: #193138;
}

.mobileFilterSubtitle {
  font-family: "Libre Franklin", sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 24px;
  color: $lightGrey;
}

.mobileFilterTitleContainer {
  flex: 1;
  display: flex;
  flex-direction: column;
  justify-content: center;
}
.mobileFilterImageContainer {
  position: relative;
  align-items: center;
  margin-left: auto;
  margin-right: 14px;
  display: flex;
}

.dateListContainer {
  .prevArrowActive {
    grid-area: arrowLeft;
    margin: auto;
    width: 22px;
    cursor: pointer;
    opacity: 1;
  }
  .prevArrowDisable {
    grid-area: arrowLeft;
    margin: auto;
    width: 22px;
    opacity: 0.3;
    cursor: not-allowed;
  }
}
