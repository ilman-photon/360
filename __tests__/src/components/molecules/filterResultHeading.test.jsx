import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import FilterResultHeading from "../../../../src/components/molecules/FilterResultHeading/filterResultHeading";

window.scrollTo = jest.fn()

describe("FilterHeading Components", () => {
  let container;
  beforeEach(() => {
    container = render(<FilterResultHeading />);
  });

  it("FilterHeading render", () => {
    expect(container).toMatchSnapshot();
  });

});
