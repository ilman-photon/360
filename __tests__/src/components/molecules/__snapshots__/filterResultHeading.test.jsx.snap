// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterHeading Components FilterHeading render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="filterContainer MuiBox-root css-0"
      >
        <div
          class="filterButtonContainer MuiBox-root css-0"
        >
          <button
            class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root filterButton css-18fc0i1-MuiButtonBase-root-MuiButton-root"
            data-testid="filterbtn"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium tuneIcon css-i4bv87-MuiSvgIcon-root"
              data-testid="TuneIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 17v2h6v-2H3zM3 5v2h10V5H3zm10 16v-2h8v-2h-8v-2h-2v6h2zM7 9v2H3v2h4v2h2V9H7zm14 4v-2H11v2h10zm-6-4h2V7h4V5h-4V3h-2v6z"
              />
            </svg>
            Filter
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium keydownIcon css-i4bv87-MuiSvgIcon-root"
              data-testid="KeyboardArrowDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <div
            class="filterChildButton MuiBox-root css-0"
          >
            Spanish
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium closeIcon css-i4bv87-MuiSvgIcon-root"
              data-testid="CloseIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </div>
          <div
            class="filterChildButton MuiBox-root css-0"
          >
            In Network
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium closeIcon css-i4bv87-MuiSvgIcon-root"
              data-testid="CloseIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </div>
        </div>
        <div
          class="MuiBox-root css-1sq0orx"
        >
          <div
            class="MuiBox-root css-jy5pi"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 filterFound css-ahj2mt-MuiTypography-root"
            >
              30 In-network providers
            </p>
          </div>
          <div
            class="MuiBox-root css-1qe922y"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1w23uea-MuiSvgIcon-root"
              data-testid="ArrowBackIosIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11.67 3.87 9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
              />
            </svg>
            <div
              class="MuiBox-root css-g9lo0d"
            >
              <div
                class="MuiBox-root css-1prvghn"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 calenderDay css-ahj2mt-MuiTypography-root"
                >
                  mon
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 calenderDay css-ahj2mt-MuiTypography-root"
                >
                  Sep 19
                </p>
              </div>
              <div
                class="MuiBox-root css-19wmdl0"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 calenderDay css-ahj2mt-MuiTypography-root"
                >
                  tue
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 calenderDay css-ahj2mt-MuiTypography-root"
                >
                  Sep 20
                </p>
              </div>
              <div
                class="MuiBox-root css-114qxfe"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 calenderDay css-ahj2mt-MuiTypography-root"
                >
                  wed
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 calenderDay css-ahj2mt-MuiTypography-root"
                >
                  Sep 21
                </p>
              </div>
              <div
                class="MuiBox-root css-cg2h46"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 calenderDay css-ahj2mt-MuiTypography-root"
                >
                  thu
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 calenderDay css-ahj2mt-MuiTypography-root"
                >
                  Sep 22
                </p>
              </div>
              <div
                class="MuiBox-root css-oacha6"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 calenderDay css-ahj2mt-MuiTypography-root"
                >
                  fri
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 calenderDay css-ahj2mt-MuiTypography-root"
                >
                  Sep 23
                </p>
              </div>
              <div
                class="MuiBox-root css-1ibu2l3"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 calenderDay css-ahj2mt-MuiTypography-root"
                >
                  sat
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 calenderDay css-ahj2mt-MuiTypography-root"
                >
                  Sep 24
                </p>
              </div>
            </div>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-x1a2ui-MuiSvgIcon-root"
              data-testid="ArrowForwardIosIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="filterContainer MuiBox-root css-0"
    >
      <div
        class="filterButtonContainer MuiBox-root css-0"
      >
        <button
          class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root filterButton css-18fc0i1-MuiButtonBase-root-MuiButton-root"
          data-testid="filterbtn"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium tuneIcon css-i4bv87-MuiSvgIcon-root"
            data-testid="TuneIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 17v2h6v-2H3zM3 5v2h10V5H3zm10 16v-2h8v-2h-8v-2h-2v6h2zM7 9v2H3v2h4v2h2V9H7zm14 4v-2H11v2h10zm-6-4h2V7h4V5h-4V3h-2v6z"
            />
          </svg>
          Filter
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium keydownIcon css-i4bv87-MuiSvgIcon-root"
            data-testid="KeyboardArrowDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div
          class="filterChildButton MuiBox-root css-0"
        >
          Spanish
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium closeIcon css-i4bv87-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </div>
        <div
          class="filterChildButton MuiBox-root css-0"
        >
          In Network
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium closeIcon css-i4bv87-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </div>
      </div>
      <div
        class="MuiBox-root css-1sq0orx"
      >
        <div
          class="MuiBox-root css-jy5pi"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 filterFound css-ahj2mt-MuiTypography-root"
          >
            30 In-network providers
          </p>
        </div>
        <div
          class="MuiBox-root css-1qe922y"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1w23uea-MuiSvgIcon-root"
            data-testid="ArrowBackIosIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11.67 3.87 9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
            />
          </svg>
          <div
            class="MuiBox-root css-g9lo0d"
          >
            <div
              class="MuiBox-root css-1prvghn"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 calenderDay css-ahj2mt-MuiTypography-root"
              >
                mon
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 calenderDay css-ahj2mt-MuiTypography-root"
              >
                Sep 19
              </p>
            </div>
            <div
              class="MuiBox-root css-19wmdl0"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 calenderDay css-ahj2mt-MuiTypography-root"
              >
                tue
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 calenderDay css-ahj2mt-MuiTypography-root"
              >
                Sep 20
              </p>
            </div>
            <div
              class="MuiBox-root css-114qxfe"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 calenderDay css-ahj2mt-MuiTypography-root"
              >
                wed
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 calenderDay css-ahj2mt-MuiTypography-root"
              >
                Sep 21
              </p>
            </div>
            <div
              class="MuiBox-root css-cg2h46"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 calenderDay css-ahj2mt-MuiTypography-root"
              >
                thu
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 calenderDay css-ahj2mt-MuiTypography-root"
              >
                Sep 22
              </p>
            </div>
            <div
              class="MuiBox-root css-oacha6"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 calenderDay css-ahj2mt-MuiTypography-root"
              >
                fri
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 calenderDay css-ahj2mt-MuiTypography-root"
              >
                Sep 23
              </p>
            </div>
            <div
              class="MuiBox-root css-1ibu2l3"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 calenderDay css-ahj2mt-MuiTypography-root"
              >
                sat
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 calenderDay css-ahj2mt-MuiTypography-root"
              >
                Sep 24
              </p>
            </div>
          </div>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-x1a2ui-MuiSvgIcon-root"
            data-testid="ArrowForwardIosIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
